src/hotspot/share/services/management.cpp

```diff
@@ -1121,6 +1121,11 @@ JVM_ENTRY(jint, jmm_GetThreadInfo(JNIEnv *env, jlongArray ids, jint maxDepth, jo
   // in the ThreadSnapshot are marked and adjusted during GC.
   ThreadDumpResult dump_result(num_threads);
 
+  // When using the new lightweight locking, we need to take
+  // a safepoint here, because the thread snapshot code calls
+  // ObjectSynchronizer::get_lock_owner() and it would potentially
+  // give wrong results when Java threads are running and
+  // entering/leaving locks while we inspect the thread stacks.
   if (maxDepth == 0) {
```

Comment:
You've removed the code that cause safepoint code path to be taken here so
this comment is no longer correct.

Also, I've verified that:
vmTestbase/nsk/monitoring/ThreadMXBean/ThreadInfo/Deadlock/JavaDeadlock005/TestDescription.java
no longer fails on my MBP13 so I agree that the `start_processing()`
call made the ZGC -Xcomp config much happier.